syntax = "proto3";

service MasterService {
    rpc ProcessImage (ImageRequest) returns (ImageResponse);   // Existing
    rpc QueryResult (ResultRequest) returns (ResultResponse); // Fetch results
    rpc GetModels (ModelRequest) returns (ModelResponse);     // List models
    rpc GetModelDetails (ModelDetailRequest) returns (ModelDetailResponse); // Model details
}

message ImageRequest {
    bytes image_data = 1;
    string user_id = 2;  // User ID for multitenancy
}

message ImageResponse {
    string request_id = 1;  // Request ID returned to the client
}

message ResultRequest {
    string request_id = 1;  // Request ID to fetch results
}

message ResultResponse {
    string status = 1;      // Processing status: PENDING, COMPLETED, FAILED
    string result = 2;      // JSON or descriptive result
}

message ModelRequest {}

message ModelResponse {
    repeated string model_names = 1; // List of available models
}

message ModelDetailRequest {
    string model_name = 1;
}

message ModelDetailResponse {
    string details = 1; // Model-specific details
}
